lib/pay_web/resolvers/mutations.ex

```diff
@@ -34,6 +35,8 @@ defmodule PayWeb.Resolvers.Mutations do
           gateway_transaction_id: refund_reference,
           user_external_id: current_user.external_id
         })
+        else
+          _ -> {:error, "Cannot refund a failed payment"}
```

Comment:
this error is misleading because it might fail for a number of reasons; not only because of a failed payment

i suggest writing a helper function like `Payments.get_successful_payment(payment_id)` which encapsulates this logic by checking for nil `gateway_transaction_id`. There you can return the appropriate error message or `{:ok, payment}` if everything went well. You can then just let it fall through in this block without matching